File: crates/weave-graph-parse/tests/fixtures/sample.php
<?php

require 'helpers.php';

interface Named {
    public function getName(): string;
}

class Greeter implements Named {
    public function getName(): string {
        return $this->name;
    }

    public function greet(): string {
        return formatName($this->getName());
    }
}

class LoudGreeter extends Greeter {
}

function formatName($name) {
    return trim($name);
}

File: crates/weave-graph-parse/tests/fixtures/sample.go
package main

import "fmt"

type Greeter struct {
	Name string
}

func (g *Greeter) Greet() string {
	return formatName(g.Name)
}

func formatName(name string) string {
	fmt.Println(name)
	return name
}

type Named interface {
	GetName() string
}
